runtime/src/bank.rs

```diff
@@ -90,11 +90,17 @@ pub struct Bank {
     last_id_queue: RwLock<LastIdQueue>,
 
     /// Previous checkpoint of this bank
-    parent: Option<Arc<Bank>>,
+    parent: RwLock<Option<Arc<Bank>>>,
 
-    /// Hash of this Bank's state. Only meaningful after freezing it via `new_from_parent()`.
+    /// Hash of this Bank's state. Only meaningful after freezing.
     hash: RwLock<Hash>,
 
+    /// Hash of this Bank's parent's state
+    parent_hash: Hash,
```

Comment:
why?

Reply:
once merge_parents() completes, hash_internal_state() needs to have the value cached...

Reply:
merge_parents() sets parent to None ;)

Reply:
I see, it's effectively because we've reordered the data after squashing. We've also removed things like accounts with zero tokens, and deleted the accounts they refer to. I can see how that'd be a lot more painful to manage than caching the result of the existing hashing algorithm.